pages/bills/[slug].tsx

```diff
@@ -1,47 +1,89 @@
+import { useMemo, useCallback } from "react";
 import type { GetServerSideProps, NextPage } from "next";
 import { useRouter } from "next/router";
 import { NextParsedUrlQuery } from "next/dist/server/request-meta";
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import nextI18nextConfig from "next-i18next.config";
 import { dehydrate, QueryClient } from "@tanstack/react-query";
 import ReactMarkdown from "react-markdown";
-import { IssueTags, Candidate, FlagSection } from "components";
-import { PoliticianResult } from "generated";
+import { BsChevronLeft } from "react-icons/bs";
 
 import {
+  IssueTags,
+  Candidate,
+  FlagSection,
   Layout,
   LoaderFlag,
   LegislationStatusBox,
   Button,
   SupportOppose,
 } from "components";
+import { CommitteeTag } from "components/PoliticianPage/CommitteesSection/CommitteesSection";
+import { SupportedLocale } from "types/global";
+
+import {
+  PoliticianResult,
+  BillBySlugQuery,
+  useBillBySlugQuery,
+} from "generated";
 
-import { BillBySlugQuery, useBillBySlugQuery } from "generated";
 import styles from "./BillBySlug.module.scss";
-import { SupportedLocale } from "types/global";
-import { serverSideTranslations } from "next-i18next/serverSideTranslations";
-import nextI18nextConfig from "next-i18next.config";
-import { CommitteeTag } from "components/PoliticianPage/CommitteesSection/CommitteesSection";
 
 const BillPage: NextPage<{ mobileNavTitle?: string }> = ({
   mobileNavTitle,
 }: {
   mobileNavTitle?: string;
 }) => {
-  const { query } = useRouter();
-  const slug = query.slug as string;
+  const router = useRouter();
+  const slug = router.query.slug as string;
 
   const { data, isLoading, error } = useBillBySlugQuery({ slug });
 
-  if (isLoading) return <LoaderFlag />;
-  if (error) return <>Error: {error}</>;
-
   const bill = data?.billBySlug;
+
   const summary =
     bill?.description || bill?.populistSummary || bill?.officialSummary;
 
+  const $supportOppose = useMemo(
+    () =>
+      bill ? (
+        <SupportOppose
+          billId={bill?.id}
+          billSlug={bill?.slug}
+          publicVotes={bill?.publicVotes}
+          usersVote={bill?.usersVote}
+        />
+      ) : null,
+    [bill]
+  );
+
+  const backAction = useCallback(() => {
+    const { referrer } = document;
+    if (
+      referrer === "" ||
+      new URL(referrer).origin !== window.location.origin
+    ) {
+      void router.push("/bills");
+    } else {
+      router.back();
+    }
+  }, [router]);
```

Comment:
nice work on this, i would have stuck a `router.back()` here and called it good :)